sqlglot/generator.py

```diff
@@ -3126,6 +3126,8 @@ def altertable_sql(self, expression: exp.AlterTable) -> str:
             actions = self.expressions(expression, key="actions", prefix="ADD COLUMNS ")
         elif isinstance(actions[0], exp.Delete):
             actions = self.expressions(expression, key="actions", flat=True)
+        elif isinstance(actions[0], exp.Select):
+            actions = "AS " + self.expressions(expression, key="actions", flat=True)
```

Comment:
why is this flat=True?

Reply:
So there is no need to flat since there is only one action?

Reply:
if it's flat, it may not be pretty printed

Reply:
Removed. Although I am not very clear how this controls the output. Thanks.

Reply:
got it. yes it will only generate one line with flat=True.